### PR TITLE
Improve dev flag

### LINKED ISSUE #162: Improve `--dev` flag
**Is your feature request related to a problem? Please describe.**

`--dev` is currently only used for csrf but we need to use it in other areas as well.

We should also check if self urls are https and if not disallow the startup!

### PR DESCRIPTION
**Pull request by zepatrik**
## Related issue

closes #162

## Proposed changes

Adds a `InsecureDevMode() bool` function to the config provider. 

## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [ ] I have read the [security policy](../security/policy)
- [ ] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation within the code base (if appropriate)
- [ ] I have documented my changes in the [developer guide](https://github.com/ory/docs) (if appropriate)

### CODE REVIEW

File: driver/configuration/provider.go [hunk 1]
```diff
@@ -78,4 +78,6 @@ type Provider interface {
 	TracingServiceName() string
 	TracingProvider() string
 	TracingJaegerConfig() *tracing.JaegerConfig
+
+	InsecureDevMode() bool
```
Comment (aeneasr) [inline driver/configuration/provider.go]: Maybe rename this to `IsInsecureDevMode()`?

File: driver/driver_default.go [hunk 2]
```diff
@@ -14,12 +14,22 @@ type DefaultDriver struct {
...
 }
 
-func NewDefaultDriver(l logrus.FieldLogger, version, build, date string) (Driver, error) {
+func NewDefaultDriver(l logrus.FieldLogger, version, build, date string, dev bool) (Driver, error) {
 	if l == nil {
 		l = logrusx.New()
 	}
 
-	c := configuration.NewViperProvider(l)
+	if dev {
```
Comment (aeneasr) [inline driver/driver_default.go]: Let's move this warning to cmd/daemon instead - otherwise we'll get a lot of warnings during testing.

Comment (zepatrik) [reply driver/driver_default.go]: The logger is not initialized in the deamon yet but that should be possible

Comment (aeneasr) [reply driver/driver_default.go]: You can call it in "ServeAll", then it will be initialized

**Comment by zepatrik** [comment]
done @aeneasr

**Comment by aeneasr** [comment]
One last place where this needs to be updated: https://github.com/ory/kratos/blob/master/cmd/daemon/serve.go#L109

**Pull request merged**